Epsilon.Host.Frontend/src/components/Performance/CompetenceGraph.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+    <ApexChart
+        :options="chartOptions"
+        :series="series"
+        height="300"
+        type="bar"
+        class="competence-graph"
+        width="490" />
+</template>
+
+<script lang="ts" setup>
+import ApexChart from "vue3-apexcharts"
+import {
+    DecayingAverageLogic,
+    DecayingAveragePerLayer,
+} from "@/DecayingAverageLogic"
+import store from "@/store"
+import { computed, onMounted } from "vue"
+import { LearningDomain, LearningDomainSubmission } from "@/api"
+
+const chartOptions = {
+    annotations: {
+        yaxis: [
+            {
+                y: 3,
+                borderColor: "red",
+                strokeDashArray: 0,
+                label: {
+                    borderColor: "red",
+                    style: {
+                        color: "#fff",
+                        background: "red",
+                    },
+                    text: "Mastery",
+                },
+            },
+        ],
+    },
+    colors: [],
+    chart: {
+        type: "bar",
+        height: 300,
+        stacked: true,
+        toolbar: {
+            show: true,
+        },
+        zoom: {
+            enabled: false,
+        },
+    },
+    dataLabels: {
+        enabled: false,
+    },
+    plotOptions: {
+        bar: {
+            horizontal: false,
+            borderRadius: 4,
+            dataLabels: {},
+        },
+    },
+    xaxis: {
+        type: "string",
+        categories: [],
+    },
+    yaxis: {
+        show: false,
+    },
+    legend: {
+        show: false,
+    },
+    fill: {
+        opacity: 1,
+    },
+    tooltip: {
+        enabled: true,
+    },
+}
+
+onMounted(() => {
+    if (store.state.domain?.columnsSet?.types != null) {
```

Comment:
This is a very long chain of properties, maybe store it in a const and then reuse that in the if statement scope. Just a way to clean things up.